app/client/lib/ethereum/observeBlocks.js

```diff
@@ -54,14 +54,42 @@ updateBalances = function() {
     // UPDATE TOKEN BALANCES
     var walletsContractsAndAccounts = EthAccounts.find().fetch().concat(walletsAndContracts);
 
-    _.each(Tokens.find().fetch(), function(token){
-        if(!token.address)
-            return;
+    
+
+    // go through all existing accounts, for each token
+    _.each(walletsContractsAndAccounts, function(account){
+
+        // Only check ENS names every N minutes
+        if (!account.ensCheck || (account.ensCheck && Date.now() - account.ensCheck > 10*60*1000)) {
+            Helpers.getENSName(account.address, (err, name, returnedAddr) => {
+                if (!err && account.address.toLowerCase() == returnedAddr){
+                    if(EthAccounts.findOne({address: account.address}))
```

Comment:
The `if`s are not necessary here. Only update with the same query would suffice.